### PR TITLE
Update Android SDK docs

### LINKED ISSUE #1543: Update Android SDK docs
Android SDK docs are outdated. Should be updated to more closely match the repo readme. Main updates:

- Now using async/await instead of callbacks - which is much neater
- Simplifying the examples, e.g. the switch chain example

### PR DESCRIPTION
**Pull request by alexandratran**
# Description

Update Android SDK docs to include coroutine examples, and convenience methods.

## Issue(s) fixed

Fixes #1543 

## Preview

https://metamask-docs-737v7zdag-metamask-web.vercel.app/wallet/how-to/use-sdk/mobile/android/

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

**Review by elefantel — DISMISSED** [review]
Looks good

**Review by joaniefromtheblock — DISMISSED** [review]
lgtm

File: wallet/how-to/use-sdk/mobile/android.md [hunk 1]
```diff
@@ -112,70 +178,91 @@ class EthereumViewModel @Inject constructor(
...
+
+The following examples use coroutines.
 
 #### Example: Get account balance
 
 The following example gets the user's account balance by calling
 [`eth_getBalance`](/wallet/reference/eth_getBalance).
+This is a [read-only request](../javascript/make-read-only-requests.md), which uses the Infura API
+if an `infuraAPIKey` is provided in the `SDKOptions`.
+We highly recommend using the Infura API to provide a seamless user experience.
```
Comment (joaniefromtheblock) [inline wallet/how-to/use-sdk/mobile/android.md:254]: I think 'highly' is not needed

```suggestion
We recommend using the Infura API to provide a seamless user experience.
```

**Review by joaniefromtheblock — APPROVED** [review]
lgtm

**Pull request merged**